File: src/test/java/com/testexcercise/ec2instance/EC2InstanceCreateTest.java
package com.testexcercise.ec2instance;

import java.util.*;
import java.util.stream.*;

import com.amazonaws.regions.*;
import com.amazonaws.services.ec2.*;
import com.amazonaws.services.ec2.model.*;

import org.testng.*;
import org.testng.annotations.*;

public class EC2InstanceCreateTest
{
    static String keyName;
    static String secGroupName = "TestSecGroup";
    static AmazonEC2 ec2client;
    
    List<Instance> instances;
    
    @BeforeClass
    public void classSetup()
    {
        keyName = "key"+UUID.randomUUID().toString();
        
        AmazonEC2ClientBuilder clientBuilder =
                AmazonEC2ClientBuilder.
                    standard().
                    withRegion(Regions.AP_SOUTHEAST_2);
        
        ec2client = clientBuilder.build();
        
        CreateSecurityGroupRequest csgreq = 
                new CreateSecurityGroupRequest().
                    withGroupName(secGroupName).
                    withDescription("Test Security Group");
        
        CreateSecurityGroupResult csgres =
                ec2client.createSecurityGroup(csgreq);
        
        
        CreateKeyPairRequest createKeyPairRequest =
                new CreateKeyPairRequest().
                    withKeyName(keyName);
        
        CreateKeyPairResult createKeyPairResult =
                ec2client.createKeyPair(createKeyPairRequest);
    }
    
    @AfterClass
    public void classTeardown()
    {
        DeleteKeyPairRequest dkpreq = 
                new DeleteKeyPairRequest().
                    withKeyName(keyName);
        ec2client.deleteKeyPair(dkpreq);
        
        DeleteSecurityGroupRequest dsgreq =
                new DeleteSecurityGroupRequest().
                    withGroupName(secGroupName);
        
        ec2client.deleteSecurityGroup(dsgreq);
    }
    
    @BeforeMethod
    public void setup()
    {
        instances = null;
    }
    
    @AfterMethod
    public void teardown()
    {
        if (instances == null)
            return;
        
        TerminateInstancesRequest terminateRequest = 
                new TerminateInstancesRequest().
                    withInstanceIds(
                            instances.stream().
                            map(x -> x.getInstanceId()).
                            collect(Collectors.toList()));
        
        ec2client.terminateInstances(terminateRequest);
    }
    
    @Parameters({ "imageId", "instanceType" })
    @Test
    public void createEC2VMInstance(
            String imageId,
            String instanceType)
    {   
        RunInstancesRequest runInstancesRequest =
                new RunInstancesRequest().
                    withImageId("ami-43874721").
                    withInstanceType("t2.micro").
                    withMinCount(1).
                    withMaxCount(1).
                    withKeyName(keyName).
                    withSecurityGroups(secGroupName);
        
        RunInstancesResult instanceRequestResult =
                ec2client.runInstances(runInstancesRequest);
        
        instances = instanceRequestResult.
                getReservation().
                getInstances();
        
        Assert.assertEquals(instances.size(), 1);
    }    
}

File: net/TestExerciseEC2/EC2InstanceCreateTest.cs
using System;
using System.Collections.Generic;
using System.Linq;

using Amazon;
using Amazon.EC2;
using Amazon.EC2.Model;

using NUnit.Framework;

namespace TestExerciseEC2
{
    public class EC2InstanceCreateTest
    {
        static String keyName;
        static String secGroupName = "TestSecGroup";
        static AmazonEC2Client ec2client;

        List<Instance> instances;

        [OneTimeSetUp]
        public void OneTimeInit()
        {
            keyName = "key" + Guid.NewGuid().ToString();

            ec2client = new AmazonEC2Client(RegionEndpoint.APSoutheast2);

            CreateSecurityGroupRequest csgreq =
                    new CreateSecurityGroupRequest(
                        secGroupName,
                        "Test Security Group");

            CreateSecurityGroupResponse csgres =
                    ec2client.CreateSecurityGroup(csgreq);


            CreateKeyPairRequest createKeyPairRequest =
                    new CreateKeyPairRequest(keyName);

            CreateKeyPairResponse createKeyPairResult =
                    ec2client.CreateKeyPair(createKeyPairRequest);
        }

        [OneTimeTearDown]
        public void ClassTeardown()
        {
            DeleteKeyPairRequest dkpreq =
                    new DeleteKeyPairRequest(keyName);

            ec2client.DeleteKeyPair(dkpreq);

            DeleteSecurityGroupRequest dsgreq =
                    new DeleteSecurityGroupRequest(secGroupName);

            ec2client.DeleteSecurityGroup(dsgreq);
        }

        [SetUp]
        public void Init()
        {
            instances = null;
        }

        [TearDown]
        public void Teardown()
        {
            if (instances == null)
                return;

            TerminateInstancesRequest terminateRequest =
                    new TerminateInstancesRequest(
                        instances.Select(ins => ins.InstanceId).ToList());

            ec2client.TerminateInstances(terminateRequest);
        }

        [Test]
        public void Create_EC2_VM_Instance()
        {
            RunInstancesRequest runInstancesRequest =
                    new RunInstancesRequest()
                    {
                        ImageId = "ami-43874721",
                        InstanceType = "t2.micro",
                        MinCount = 1,
                        MaxCount = 1,
                        KeyName = keyName,
                        SecurityGroups = new List<string>() { secGroupName }
                    };

            RunInstancesResponse instanceRequestResult =
                    ec2client.RunInstances(runInstancesRequest);

            instances = instanceRequestResult.Reservation.Instances;

            Assert.That(instances.Count, Is.EqualTo(1));
        }
    }
}
